readthedocs/projects/views/mixins.py

```diff
@@ -44,3 +47,46 @@ def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         context[self.project_context_object_name] = self.get_project()
         return context
+
+
+class ProjectImportMixin:
+
+    """Helpers to import a Project."""
+
+    def import_project(self, project, tags, request):
+        """
+        Import a Project into Read the Docs.
+
+        - Add the user from request as maintainer
+        - Set all the tags to the project
+        - Send Django Signal
+        - Trigger initial build
+        """
+        project.users.add(request.user)
+        for tag in tags:
+            project.tags.add(tag)
+
+        # TODO: this signal could be removed, or used for sync task
+        project_import.send(sender=project, request=request)
+
+        self.trigger_initial_build(project, request.user)
+
+    def trigger_initial_build(self, project, user):
+        """
+        Trigger initial build after project is imported.
+
+        :param project: project's documentation to be built
+        :returns: Celery AsyncResult promise
+        """
+
+        update_docs, build = prepare_build(project)
+        if (update_docs, build) == (None, None):
+            return None
+
+        from readthedocs.oauth.tasks import attach_webhook
+        task_promise = chain(
+            attach_webhook.si(project.pk, user.pk),
+            update_docs,
+        )
```

Comment:
Do we need to chain these? Seems like we still want to update the docs even if webhook setup fails, and they don't depend on each other in any real way.

Reply:
We do a chain just to follow an order. If `attach_webhook` fails, it handles it by itself, set a message to the user and the build process continue. It does not block the update of the docs.

Anyways, this logic is not something I want to change/refactor in this PR because it's something bigger than this.